### PR TITLE
print-summary: support URL path to summary file

### PR DESCRIPTION
**Pull request by jlebon**
It's just more convenient to be able to just paste in the summary URL
directly without having to download it first.

### CODE REVIEW

File: print-summary [hunk 1]
```diff
@@ -22,8 +22,16 @@ parser.add_argument("path", help="Summary path as file",
...
 args = parser.parse_args()
 
-with open(args.path) as f:
-    bytedata = GLib.Bytes.new(f.read())
+if args.path.startswith("http://") or args.path.startswith("https://"):
+    r = requests.get(args.path)
+    if r.status_code != requests.codes.ok:
+        print("ERROR: couldn't fetch summary file:")
+        print(r.headers)
+        print(r.text)
```
Comment (cgwalters) [inline print-summary]: Move the first `print` here and make it `fatal`?

Comment (jlebon) [reply print-summary]: Oh right, of course it has to be fatal!

File: print-summary [hunk 2]
```diff
@@ -25,9 +25,10 @@ args = parser.parse_args()
 if args.path.startswith("http://") or args.path.startswith("https://"):
     r = requests.get(args.path)
     if r.status_code != requests.codes.ok:
-        print("ERROR: couldn't fetch summary file:")
         print(r.headers)
         print(r.text)
+        print("ERROR: couldn't fetch summary file")
+        sys.exit(1)
```
Comment (cgwalters) [inline print-summary]: There's a helper function already defined called `fatal` that I cargo cult around.

Comment (jlebon) [reply print-summary]: Ahh gotcha. Switched to `fatal` and changed error style to match the rest of the repo.

**Pull request merged**